src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java

```diff
@@ -52,7 +54,7 @@ public ConstraintSpec getConstraintSpec() {
           .types(APIVariableType.NUMBER, APIVariableType.DATE, APIVariableType.INTEGER)
           .description("Variable must be a number or date.")
         .element("xAxisVariable")
-          .types(APIVariableType.NUMBER, APIVariableType.DATE, APIVariableType.INTEGER)
+          .shapes(APIVariableDataShape.ORDINAL, APIVariableDataShape.CONTINUOUS)
```

Comment:
what happens with binning if a non-date, non-numeric ordinal var is used on the x axis? Like a var describing how sick you are with "not sick", "sick", "really sick"?

Reply:
i imagine wed ignore the bin width specified and issue a warning.. doesnt make sense to try to bin that, at least in this context. thatd be a derived variable. but i think that might be plot.data business, so ill make a note to add some unit tests etc there for the case.